openPowerlink VxWorks implementation

jbaumgar opened the pull request:
Please pull changes for VxWorks implementation

Commit:
Add VxWorks high-resolution timer library

Commit:
Add VxWorks timer implementations

Commit:
Add VxWorks shared buffer implementation

Commit:
Add Edrv for VxWorks MUX interface

Commit:
Add VxWorks support in several stack files

Commit:
Add VxWorks X86 MN demo application

Commit:
Add VxWorks documentation

systec-dk commented on EplStack/EplObdCdc.c:
```diff
@@ -273,7 +273,11 @@ DWORD           dwErrno;
+#if (TARGET_SYSTEM == _VXWORKS_)
+    CdcInfo.m_Handle.m_hCdcFile = open(pszCdcFilename_p, O_RDONLY, 0666);
+#else
     CdcInfo.m_Handle.m_hCdcFile = open(pszCdcFilename_p, O_RDONLY | O_BINARY, 0666);
```
Does O_BINARY not exist in VxWorks? If it does not exist, it seems to be better to define it to 0.

jbaumgar replied:
Yes, we should define it.

systec-dk commented on Include/global.h:
```diff
@@ -912,8 +959,6 @@
         #define TARGET_SYSTEM       _NO_OS_
         #define DEV_SYSTEM          _DEV_NIOS2_
-
-    #else
```
Why do you remove the #else case?

jbaumgar replied:
Sorry, was accidentally removed!

Commit:
Add some bugfixes and cleanups

- Add missing project files
- Add missing else statement in global.h
- Add definition of O_BINARY for vxworks in EplObdCdc.c

Commit:
Adjust VxWorks task priorities

- Changed priorities for openPOWERLINK tasks in EplCfg.h
- Changed priority increase/decrease to 5 in ShbIpc-VxWorks.c
- Added adjustment of system task priorities in demo-main.c
- Added documentation of priorities in vxworks.txt

systec-dk commented:
Merged to master branch
